### PR TITLE
fix(core): target project locator should handle tsconfig paths fully

### PR DESCRIPTION
**Pull request by meeroslav**
## Current Behaviour
The `TargetProjectLocator` does not match deep imports (e.g. `@nrwl/tao/src/utils/app-root`) with `tsconfig` wildcard paths (e.g. `@nrwl/tao/*`). 

As a consequence, if we have the same package (e.g. `tao`) locally and installed as an npm dependency, we need to run `typescript resolver` matching before `npm` matching to make sure we target the local one. (See https://github.com/nrwl/nx/pull/6030)

This is an expensive operation, causing slowdowns on `linter`. (See https://github.com/nrwl/nx/pull/5576)

## Expected Behaviour
The `TargetProjectLocator` should match deep imports with `tsconfig` wildcard paths.

NPM package resolving should run before typescript matching, to avoid unnecessary slowdown.

The typescript resolver should probably be removed altogether.

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/workspace/src/core/target-project-locator.ts [hunk 1]
```diff
@@ -60,15 +60,73 @@ export class TargetProjectLocator {
...
+      normalizedImportExpr,
+      filePath
+    );
+    if (resolvedProject) {
+      return resolvedProject;
+    }
+
+    // TODO(meeroslav): this block should be removed as it's going around typescript paths
+    // unless we want to support local JS projects
+    const importedProject = this.sortedWorkspaceProjects.find((p) => {
```
Comment (meeroslav) [inline packages/workspace/src/core/target-project-locator.ts:92]: This makes sense if we want to support JS projects or making sure for imports to be validated even if path is missing in `tsconfig`.

File: packages/workspace/src/core/target-project-locator.ts [hunk 2]
```diff
@@ -60,15 +60,73 @@ export class TargetProjectLocator {
...
 
+    // try to find npm package before using expensive typescript resolution
+    const npmProject = this.findNpmPackage(normalizedImportExpr);
+    if (npmProject) {
+      return npmProject;
+    }
+
+    // TODO(meeroslav): this block is probably obsolete
+    // and existed only because of the incomplete `paths` matching
+    // if import cannot be matched using tsconfig `paths` the compilation would fail anyway
```
Comment (meeroslav) [inline packages/workspace/src/core/target-project-locator.ts:81]: After fixing path matcher, I don't see the scenario when this might produce result.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
We found an issue where the paths in `tsconfig.base.json` are not normalized.

**Comment by meeroslav** [comment]
@FrozenPandaz Please have another check

**Review by FrozenPandaz — APPROVED** [review]
Looks good

**Pull request merged**